### PR TITLE
Implement proper test runner using scons

### PR DESCRIPTION
**Pull request by michal-clapinski**

### CODE REVIEW

File: CIScripts/Build/BuildFunctions.ps1 [hunk 1]
```diff
@@ -294,21 +225,8 @@ function Invoke-ProductUnitTests {
...
-        $TestRes | ForEach-Object {
-            if (0 -ne $_) {
-                throw "Running tests failed"
-            }
-        }
-
-        $Env:Path = $backupPath
+        Invoke-NativeCommand -ScriptBlock {
+            scons -j $Env:BUILD_THREADS --opt=$BuildMode @Tests
+        } | Out-Null
```
Comment (jacekchalupka) [inline CIScripts/Build/BuildFunctions.ps1]: Won't the Out-Null prevent logs from being written?

File: CIScripts/Build/BuildFunctions.ps1 [hunk 2]
```diff
@@ -218,72 +203,18 @@ function Copy-DebugDlls {
...
-            "$BaseTestPrefix/queue_task_test.exe"
-            "$BaseTestPrefix/patricia_test.exe"
-            "$BaseTestPrefix/label_block_test.exe"
-            "$BaseTestPrefix/factory_test.exe"
-            "$BaseTestPrefix/index_allocator_test.exe"
-            "$BaseTestPrefix/dependency_test.exe"
-            "$BaseTestPrefix/bitset_test.exe"
-
-            #"src/contrail-common/base:test"
+            "src/contrail-common/base:test"
```
Comment (jacekchalupka) [inline CIScripts/Build/BuildFunctions.ps1]:
```suggestion
            'src/contrail-common/base:test',
```
It works without comma, but everywhere else we use them. And please use single quote when double quote is not needed.

File: CIScripts/Build/BuildFunctions.ps1 (hunk 1 above)
Comment (michal-clapinski) [reply CIScripts/Build/BuildFunctions.ps1]: No

**Comment by michal-clapinski** [comment]
recheck no bug

**Comment by michal-clapinski** [comment]
recheck no bug

**Pull request merged**